icu4c/source/i18n/plurrule.cpp

```diff
@@ -1881,8 +1881,8 @@ void FixedDecimal::adjustForMinFractionDigits(int32_t minFractionDigits) {
 
 double FixedDecimal::getPluralOperand(PluralOperand operand) const {
     switch(operand) {
-        case PLURAL_OPERAND_N: return source;
-        case PLURAL_OPERAND_I: return static_cast<double>(intValue);
+        case PLURAL_OPERAND_N: return (exponent == 0 ? source : source * pow(10, exponent));
+        case PLURAL_OPERAND_I: return (double) longValue();
```

Comment:
@sffc is `static_cast<double>` or `(double)` better to use in C++ here?

Reply:
I prefer `static_cast<double>` but Markus prefers `(double)`.  The C++ standard is to use `static_cast<double>`.

Reply:
I do not have a preference on the casting, but I am glad to see the exponent support added